### PR TITLE
Don't load Flow if a .flowconfig cannot be found

### LINKED ISSUE #81: Do not require global Flow
We do not use Flow in every project and even the ones where we do, it's not a global dependency - it's installed local every time. Thus, we get this error a lot when opening any project. Please consider making the global check fail gracefully.

screen shot 2017-02-14 at 9 26 14 am

### PR DESCRIPTION
**Pull request by orta**
Before loading up the extension it will do a check from the workspace root. To my knowledge there is no way to declare a `.flowconfig` in a subfolder of your project, so this should be fine. ( @gabelevi can you confirm? )

@Gozala wanna give it a review?

fixes #81

### CODE REVIEW

File: lib/flowMain.js [hunk 1]
```diff
@@ -29,27 +29,35 @@ const languages = [
...
 		return
 	}
-	global.vscode = vscode
 
-	setupLogging()
-	checkNode()
-	checkFlow()
+	findFlowConfigDir(vscode.workspace.rootPath).then(configPath => {
+		if (!configPath) {
+			console.log(`Not starting Flow: Could not find a .flowconfig after a backwards search from ${vscode.workspace.rootPath}.`)
```
Comment (orta) [inline lib/flowMain.js]: This ends up in the vscode developer tools web inspector, which is better than nothing IMO.

**Comment by gabelevi** [comment]
I think sometimes people have something a `src/` directory and then have `src/.flowconfig`. So I'm not sure if calling `findFlowConfigDir()` from the root will be a complete check.

**Comment by orta** [comment]
That's a good point, yeah. 

I think the feature is important, think it's worth globing all folders ( removing any node_modules ) and look through all of those too?

**Comment by Gozala** [comment]
> Before loading up the extension it will do a check from the workspace root. To my knowledge there is no way to declare a .flowconfig in a subfolder of your project, so this should be fine. ( @gabelevi can you confirm? )
> 
> @Gozala wanna give it a review?

- This seems to conflict with #84 as it treats buffers / tabs with `@flow` pragma in `js` mode as `flowtype.org/try` editors. I often times fire up new vscode window and unsaved tab to prototype things up & I don't even have workspace setup.

- What if `.flowconfig` is added afterwards ? It seems that you'd need to reload window to get the plugin loaded.
- Also I often do have `.flowconfig` under project subdir, sometimes even multiple one of them for example: `./server/.flowconfig` and `./client/.flowconfig`.

I am not entirely sure what exactly happens during activation, but as far as I can tell flow server isn't actually started until qualified file is loaded, so I'm not exactly sure what's being fixed here.

> 
> fixes #81

**Comment by orta** [comment]
The issue is that it does an annoying popup in any workspace the first time you click on a JavaScript file when there's no Flow connection, perhaps tying that popup to not loading up the server.

**Comment by orta** [comment]
I think I could replace this with a much simpler "is this a typescript project?" check (which is the only time I don't want to be running Flow basically)

**Comment by orta** [comment]
What I ended up doing instead of this PR, for my own setup:

* In my personal VS Code settings, turn off flow
* In my flow repos, add a `.vscode/settings.json` to turn it on

**Comment by IljaDaderko** [comment]
This issue happens a lot when `"flow.useNPMPackagedFlow": true` is present, it shows flow error in projects that don't have `.flowconfig` file.

**Comment by karlhorky** [comment]
@orta @gabelevi Is it possible this feature could still be implemented in some form? If it's not bulletproof, maybe behind an option?

**Comment by gaastonsr** [comment]
This is so needed.

**Comment by orta** [comment]
Maybe it could be an option like how [prettier-vscode does](https://github.com/prettier/prettier-vscode/blob/master/package.json#L69-L74) - which would only make it run if there was a flowconfig in the root. I'd recommend implementing it yourself, so maybe I should just close this PR rather than have it hanging around stopping other contributors.

**Pull request closed without merging**